### ISSUE TITLE
How to change the return value of credential authorize

### ISSUE THREAD

**Issue by kwakky1** [nextauthjs/next-auth#7236]
### Question 💬

I added id and mobile to the return value of credential authorize, but they are not added to the session value received from the callback session.

i already set type in session

import NextAuth, { DefaultSession } from "next-auth";

declare module "next-auth" {
  /**
   * Returned by `useSession`, `getSession` and received as a prop on the `SessionProvider` React Context
   */
  interface Session {
    user: {
      id: string;
      code: string;
      mobile: string;
      social?: string;
    } & DefaultSession["user"];
  }
}

### How to reproduce ☕️

CredentialsProvider({
      id: "noMember",
      type: "credentials",
      name: "noMember",
      credentials: {
        name: {
          label: "name",
          type: "text",
          placeholder: "이름을 입력하세요.",
        },
        phone: {
          label: "phone",
          type: "text",
          placeholder: "휴대전화번호를 입력하세요.",
        },
      },

      async authorize(credentials, req): Promise<any> {
        console.log("noMember");
        if (!credentials) return null;

        const { name, phone } = credentials;

        return {
          id: "0",
          name,
          mobile: phone,
        };
      },
    }),
    
    
    
    async session({ session, token, user }: any) {
    console.log(session) // this session don't have id and mobile
    }

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by balazsorban44** [comment]
it's inferred based on the `credentials` property

https://github.com/nextauthjs/next-auth/blob/7a8c0068c4c42761b54eff496f6cb8820b54b5d1/packages/next-auth/src/providers/credentials.ts#L16-L20

You don't need to manually type anything.

If you don't actually mean TypeScript types but runtime values, the credentials provider's values are first sent to the `jwt` callback.

**Comment by park-jemin** [comment]
More on this here: 
https://next-auth.js.org/providers/credentials#example---username--password
https://next-auth.js.org/configuration/callbacks#jwt-callback 

TLDR: you need to forward those fields from the user object or JWT (after first sending them to the `jwt` callback to have them accessible in the `Session` object. 

```js
callbacks: {
  ...
  session({ session, user }) {
    if (session.user && user) {
       session.user.id = user.id
       session.user.mobile = user.mobile
    }
    return session
  } 
}
```
